test/unit/enumerator_builder_test.rb

```diff
@@ -55,6 +55,11 @@ class EnumeratorBuilderTest < ActiveSupport::TestCase
       enumerator_builder(wraps: 0).build_throttle_enumerator(nil, throttle_on: -> { false }, backoff: 1)
     end
 
+    test_builder_method(:build_csv_enumerator) do
+      enum = enumerator_builder(wraps: 0).build_csv_enumerator(CSV.new("test"), cursor: nil)
+      assert_instance_of(Enumerator::Lazy, enum)
```

Comment:
I wanted to validate that `JobIteration::CsvEnumerator` is being used was being used. Do you think I should add it as a separate test?

Reply:
Probably. This `test_builder_method` method should return the enumerator itself, as it generates a test through meta-programming. If you want to test something in addition, adding a separate test seems better.

Reply:
The EnumeratorBuilder methods are straightforward enough that you may not need a specific assertion. I'm not sure what this one really brings us for example, a Lazy Enumerator could be anything really. In any case if you really care about asserting this, another test would be preferable, those are really about the wrapping feature.

> `test_builder_method` method should return the enumerator itself

Technically since it sets up a mocha expectation that is just expected to be triggered during the block, we don't really care about the return value, it does check that this enumerator isn't wrapped.